Backends/Wayland/port.lisp

```diff
@@ -0,0 +1,332 @@
+                            port-surface
+                            (make-instance 'wayland-xdg-toplevel))))
+
+    ;; Make one more round trip to handle events triggered by bindings above
+    ;; before the main event loop is started
+    (roundtrip port)
+    (wlc:wl-surface-commit (wayland-port-window port))))
+
+(defmethod initialize-instance :after ((port wayland-port) &key)
+  (with-slots (frame-managers) port
+    (push (apply #'make-instance 'wayland-frame-manager :port port nil)
+          frame-managers))
+  (initialize-wayland port)
+  (make-graft port)
+  (format t "Starting Event Loop~%")
+  (when clim-sys:*multiprocessing-p*
+    (flet ((wayland-port-event-loop ()
+             (loop
+               (with-simple-restart
+                   (restart-event-loop
+                    "Restart Wayland event loop.")
+                 (loop (process-next-event port))))))
+      (setf (port-event-process port)
+            (clim-sys:make-process
+             #'wayland-port-event-loop
+             :name (format nil "~S's event process." port))))))
+
+
+(defmethod destroy-port :before ((port wayland-port))
+  (alx:when-let ((display (wayland-port-display port)))
+    (with-accessors ((wm-base %wayland-wm-base)
+                     (window wayland-port-window)
+                     (compositor wayland-port-compositor)
+                     (registry wayland-port-registry))
+        port
+      (loop for wayland-object in (list wm-base window compositor registry)
+            when wayland-object
+              do (wlc:wayland-destroy wayland-object)))
+
+    (setf (wayland-port-display port) nil)
+    (wlc:wl-display-disconnect display)))
+
+(defmethod port-force-output ((port wayland-port))
+  (format t "PORT-FORCE-OUTPUT~%")
+  ;; (alx:when-let* ((graft (graft port))
+  ;;                 (mirror (sheet-mirror graft))
+  ;;                 (egl-display (wayland-egl-mirror-display mirror))
+  ;;                 (egl-surface (wayland-egl-mirror-surface mirror)))
+  ;;   (format t "egl swap buffers~%")
+  ;;   (egl:swap-buffers egl-display egl-surface))
+  (map-over-grafts #'%graft-force-output port)
+  (wlc:wl-surface-commit (wayland-port-window port)))
+
+;;; Port event handling
+(defmethod handle-event
+    ((port wayland-port) (event window-configuration-event))
+
+  (let ((sheet (event-sheet event))
+        (native-region (window-event-native-region event)))
+    (format t "window-configuration-event ~S ~S~%" sheet native-region)))
+
+(defmethod port-set-mirror-geometry
+    ((port wayland-port) (sheet mirrored-sheet-mixin) region)
+  (alx:when-let ((mirror (sheet-direct-mirror sheet)))
+    (with-bounding-rectangle* (x1 y1 x2 y2 :width w :height h) region
+      (with-bounding-rectangle* (ox1 ox2 oy1 oy2) (sheet-mirror-geometry sheet)
+        (let ((window (window mirror)))
+          (when (some #'/=
+                      (list x1 y1 x2 y2)
+                      (list ox1 oy1 ox2 oy2))
```

Comment:
if wayland doesn't accept real numbers (i.e width as a ratio or a float), then you should check these here I think.